From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Mon, 20 Mar 2023 21:23:36 +0100
Subject: st/icon-thene: Disconnect from settings signals on disposal

Icon themes can be created by extensions and thus are likely destroyed on
extension unload, so we must disconnect from StSettings signals to avoid
accessing to invalid memory.

Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/gnome-shell/+bug/2012021
Origin: https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2711
---
 src/st/st-icon-theme.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/st/st-icon-theme.c b/src/st/st-icon-theme.c
index a7cbd0d..b4dd489 100644
--- a/src/st/st-icon-theme.c
+++ b/src/st/st-icon-theme.c
@@ -478,8 +478,8 @@ st_icon_theme_init (StIconTheme *icon_theme)
   icon_theme->pixbuf_supports_svg = pixbuf_supports_svg ();
 
   settings = st_settings_get ();
-  g_signal_connect (settings, "notify::gtk-icon-theme",
-                    G_CALLBACK (theme_changed), icon_theme);
+  g_signal_connect_object (settings, "notify::gtk-icon-theme",
+                           G_CALLBACK (theme_changed), icon_theme, 0);
   update_current_theme (icon_theme);
 }
 
